lib/Sema/ConstraintSystem.h

```diff
@@ -3444,17 +3467,24 @@ class ConstraintSystem {
     ConstraintLocator *DefaultableBinding = nullptr;
 
     PotentialBinding(Type type, AllowedBindingKind kind,
-                     ConstraintKind bindingSource,
+                     ConstraintLocator *bindingSource,
+                     ConstraintKind BindingSourceKind,
```

Comment:
I think it might be worth to record a `Constraint*` here directly because it's effectively the same storage cost for additional information. I'm going to create a separate PR for that because I also need that in couple more places...

Reply:
Sure 👍 
If you don't mind I can do it and open a PR if is just that simple change if you don't mind :)

Reply:
Unfortunately this is going to be pretty involved, I'm hoping to get PR ready later today or tomorrow. The reason why it's harder is because we have "holes" now which do not have constraint as a source.